common/types/userInLobbyType.js

```diff
@@ -4,4 +4,6 @@ import type { UserReadyStateT } from "./userReadyStateType";
 export type UserInLobbyT = {|
   username: string,
   state: UserReadyStateT,
+  writtenCharacter?: string,
+  assignedCharacter?: string,
```

Comment:
I think we can do a better typing.
Ideally we should make undefined states impossible to represent.
I was thinking something like:
```
type userNotReadyInLobby = {| username |};
type userReadyInLobby = {| username, writtenCharacter |}
type userInLobby = userReadyInLobby | userNotReadyInLobby;
```
Works for now though.

Edit: I just realized you did something similar